google/cloud/bigtable/internal/async_bulk_apply_test.cc

```diff
@@ -46,15 +46,42 @@ class AsyncBulkApplyTest : public bigtable::testing::TableTestFixture {
             internal::kBigtableTableAdminLimits)),
         idempotent_mutation_policy_(
             bigtable::DefaultIdempotentMutationPolicy()),
-        metadata_update_policy_("my_tqble", MetadataParamTypes::NAME) {}
+        metadata_update_policy_("my_table", MetadataParamTypes::NAME) {}
+
+  void SimulateIteration() {
+    cq_impl_->SimulateCompletion(true);
+    // state == PROCESSING
+    cq_impl_->SimulateCompletion(true);
+    // state == PROCESSING, 1 read
+    cq_impl_->SimulateCompletion(false);
+    // state == FINISHING
+    cq_impl_->SimulateCompletion(true);
+  }
+
+  void SimulateIterationThatRetries() {
```

Comment:
`SimulateIterationAndRetry()` maybe?  Though in general I think the more "unrolled" tests are the better (you do not want your test to get so complicated that they need test themselves 😬 ), so maybe just not have this function at all.  Your call.

Reply:
Ack. My thinking for `SimulateIteration` was that if we change the exact way `AsyncBulkApply` uses the `CompletionQueue` (i.e. add/remove an operation behind the scenes), we would only need to update one place in the test.

I unrolled `SimulateIterationThatRetries` into:
 ```c++
 SimulateIteration();
// Start backoff policy timer before a retry
 cq_impl_->SimulateCompletion(ok)
 ```